include/ncengine/ui/ImGuiUtility.h

```diff
@@ -29,6 +29,17 @@ constexpr auto g_minAngle = -g_maxAngle;
 constexpr auto g_minScale = 0.0001f;
 constexpr auto g_maxScale = 1000.0f;
 
+/** @brief Create a top-level window. */
+template<class F>
+void Window(const char* label, F&& drawContents);
+
+/** @brief Create a child window. */
+template<class F>
+void ChildWindow(const char* label, F&& drawContents);
```

Comment:
Are there any preconditions in these methods? i.e. Does a ChildWindow require a Window to be created? If so, will you list them in the comments?

Reply:
You can make a child window without a parent. I think there is some ephemeral debug window always lurking under the surface, and you get put inside that. You can also make a normal window from inside a child window without issues, although it doesn't become a child. Its fairly forgiving.

But now that you bring it up, all of these functions need to be called only from within a ui callback, so like an `IUI` implementation or a DrawUI handler. I wonder if there's a way to make some kind of file/scoped comment that doxygen would understand.

Reply:
With a @file comment, I can add a @note comment to the the whole file.